src/components/instant-apps-filter-list/instant-apps-filter-list.tsx

```diff
@@ -0,0 +1,797 @@
+            min={min}
+            max={max}
+            scale="s"
+            lang={this.locale ?? 'en'}
+            overlay-positioning="fixed"
+            layout="vertical"
+            range
+          />
+          <calcite-action onClick={this.handleResetDatePicker.bind(this, expression, layerId)} icon="reset" text={this.messages.resetDatePicker} scale="s" />
+        </div>
+      </label>
+    ) : null;
+  }
+
+  renderFooter(): VNode {
+    return (
+      <div class={CSS.footer} slot="footer">
+        <calcite-button appearance="outline" width="full" disabled={this.disabled} onClick={this.handleResetFilter.bind(this)}>
+          {this.messages.resetFilter}
+        </calcite-button>
+      </div>
+    );
+  }
+
+  renderFullFooter(): VNode {
+    const closeText = this.closeBtnText != null ? this.closeBtnText : this.messages.close;
+    return (
+      <div class={CSS.footer} slot="footer">
+        <calcite-button appearance="outline" width="half" disabled={this.disabled} onClick={this.handleResetFilter.bind(this)}>
+          {this.messages.resetFilter}
+        </calcite-button>
+        <calcite-button appearance="solid" width="half" kind="brand" onClick={this.closeBtnOnClick?.bind(this)}>
+          {closeText}
+        </calcite-button>
+      </div>
+    );
+  }
+
+  handleResetDatePicker(expression: Expression, layerId: string): void {
+    const datePicker = this.panelEl.querySelector(`[id='${expression.definitionExpressionId}']`) as HTMLCalciteInputDatePickerElement;
+    resetDatePicker(datePicker);
+    expression.active = false;
+    expression.definitionExpression = undefined;
+    this.generateOutput(layerId);
+  }
+
+  initInput(layerId: string, expression: Expression): VNode | undefined {
+    const { type } = expression;
+    if (type === 'string' || type == 'coded-value') {
+      return this.renderCombobox(layerId, expression);
+    } else if (type === 'number' || type == 'range') {
+      return this.renderNumberSlider(layerId, expression);
+    } else if (type === 'date') {
+      return this.renderDatePicker(layerId, expression);
+    }
+    return;
+  }
+
+  async initExpressions(): Promise<void> {
+    if (this.layerExpressions == null) return;
+    const tmpLE = this.layerExpressions;
+    for (let i = 0; i < tmpLE?.length; i++) {
+      const { id } = tmpLE[i];
+      const expressions = tmpLE[i].expressions;
+      for (let j = 0; j < expressions?.length; j++) {
+        expressions[j].definitionExpressionId = `${id}-${j}`;
+        if (expressions[j].active == null && expressions[j].definitionExpression != null) {
+          expressions[j].active = false;
+        }
+        await this.setInitExpression(id, expressions[j]);
+      }
+    }
+
+    this.layerExpressions = [...tmpLE];
+  }
+
+  async getMessages(): Promise<void> {
+    const messages = await getLocaleComponentStrings(this.el);
+    this.messages = messages[0] as typeof FilterList_T9n;
+  }
+
+  handleResetFilter(): void {
```

Comment:
some of these methods are fairly large, should we break this up into separate methods with names that describe each part is doing so it's easier to jump back in the code in the future?